src/app/models/treeoperations.cpp

```diff
@@ -42,23 +42,21 @@ void TreeOperations::getDatabases(std::function<void (RedisClient::DatabaseList)
 
     if (m_connection->mode() != RedisClient::Connection::Mode::Cluster) {
         //detect all databases
-        RedisClient::Response scanningResp;
-        int dbIndex = (availableDatabeses.size() == 0)? 0 : availableDatabeses.lastKey() + 1;
-
-        while (true) {
-            try {
-                scanningResp = m_connection->commandSync("select", QString::number(dbIndex));
-            } catch (const RedisClient::Connection::Exception& e) {
-                throw ConnectionsTree::Operations::Exception(QObject::tr("Connection error: ") + QString(e.what()));
-            }
+        RedisClient::Response keyspaceResp;
+        int dbIndex = 0;
+        int idxCount = 0;
 
-            if (!scanningResp.isOkMessage())
-                break;
+        keyspaceResp = m_connection->commandSync("info", QString("keyspace"));
```

Comment:
Here you duplicating code from https://github.com/uglide/qredisclient/blob/4348566a88f627ef9a125653f89a43a1fd70b0a4/src/qredisclient/connection.cpp#L564-L565

It's better to add limit for db scanning and use it in `select` scanning loop

Reply:
Haha. I just reread code and understand that I did exactly the same but with a lot of useless logic.. :) Friday, evening.

I think that the best solution for current version is to do an ability to configure this in server configuration.
I mean some flag which say that if you select it, the utility will try to find all exists databases and show them.
Or if it not selected, the utility will just use exists database (or db0 if there is no any keys).
And as future improvement it could be possible to do a selection of which databases you want to work with and set of indexes.